STModCellular.cpp

```diff
@@ -73,37 +124,50 @@ nsapi_error_t STModCellular::soft_power_on() {
         _at->write_int(2);
         _at->write_int(2);
         _at->cmd_stop_read_resp();
-
         err = _at->get_last_error();
+        _at->unlock();
+
         if (err == NSAPI_ERROR_OK) {
             tr_debug("Flow control turned ON");
         } else {
             tr_error("Failed to enable hw flow control");
         }
-
-        _at->restore_at_timeout();
-        _at->unlock();
     }
 #endif
+
+    wait_ms(500);
+
+    _at->lock();
+    /*  Verify Flow Control settings */
```

Comment:
Are we actually verifying anything here ? Maybe shall this be done only if `AT_DEBUG` is set ?

Reply:
Makes sense, will do !